tests/ui/type_id_on_box.rs

```diff
@@ -0,0 +1,20 @@
+//@run-rustfix
+
+#![warn(clippy::type_id_on_box)]
+
+use std::any::{Any, TypeId};
```

Comment:
I'd like to see a test where the type is behind a type alias, just to be sure we are  privy to possible interactions.

Reply:
I added a few more tests. Not sure if this is how you meant it though